src/clj_kondo/impl/analyzer.clj

```diff
@@ -1604,6 +1604,17 @@
                              (node->line filename f
                                          :invalid-arity
                                          (linters/arity-error nil fn-name arg-count fixed-arities varargs-min-arity))))))))))
+    (when (and (not (utils/linter-disabled? ctx :two-argument-reduce))
+               (= 'reduce resolved-as-name)
+               (or (= 'clojure.core resolved-namespace)
+                   (= 'clojure.cljs resolved-namespace))
```

Comment:
resolved-namespace is the resolved-namespace of the higher order function, not of reduce itself

Reply:
we can probably just add extra arguments to add the resolved-ns + resolved-name from analyze-call if we don't have them here